utils/gyb_syntax_support/__init__.py

```diff
@@ -141,5 +143,28 @@ def dedented_lines(description):
     return textwrap.dedent(description).split('\n')
 
 
+def hash_syntax_node(node):
+    # Hash into the syntax name and serialization code
+    result = hash((node.name, str(get_serialization_code(node.syntax_kind))))
```

Comment:
Oops sorry, missed a nitpick, you don't have to convert the integer to a string in order to hash it.
Also no need for `str(child.is_optional)` and `str(token.serialization_code)` below.